### PR TITLE
Husan's solution

### PR DESCRIPTION
**Pull request by husan-eshonqulov**
Here I confirm that I:

- [x] ran `yarn test` before pushing changes.
- [x] wrote myself and understood all the code.
- [x] checked cleanliness of my code after finishing the implementation.

### CODE REVIEW

File: resources.js [hunk 1]
```diff
@@ -1,15 +1,86 @@
...
+    const curRatingsLength = curRatings.length;
+    const curRatingsSum = curRatings.reduce((acc, curValue) => acc + curValue);
+    const avrRating = curRatingsSum / curRatingsLength;
+    
+    return avrRating;
+  }
+
+  toggleCheckOutStatus(){
+    this._isCheckedOut = !this._isCheckedOut;
+    return this._isCheckedOut;
```
Comment (bahriddin) [inline resources.js]: This wasn't required in the README but I don't mind.

File: resources.js [hunk 2]
```diff
@@ -1,15 +1,86 @@
...
+  }
+  
+  get ratings(){
+    return this._ratings;
+  }
+
+  getAverageRating(){
+    const curRatings = this._ratings;
+    const curRatingsLength = curRatings.length;
```
Comment (bahriddin) [inline resources.js]: Not a big issue but these 2 are not adding great value to the cleanliness but adding even more variables. I'd recommend to remove.

**Review by bahriddin — CHANGES_REQUESTED** [review]

File: resources.js (hunk 2 above)
Comment (husan-eshonqulov) [reply resources.js]: Ok. I will remove them later.
I tried to write cleaner code. But I understand you.

File: resources.js (hunk 1 above)
Comment (husan-eshonqulov) [reply resources.js]: Ok. I thought the function should toggle the CheckOut status opposite, shouldn't it?

Comment (bahriddin) [reply resources.js]: Yes, it should. I mean returning it's value is redundant.

File: resources.js [hunk 3]
```diff
@@ -1,15 +1,84 @@
...
+  
+  get ratings(){
+    return this._ratings;
+  }
+
+  getAverageRating(){
+    const curRatingsSum = this._ratings.reduce((acc, curValue) => acc + curValue);
+    const avrRating = curRatingsSum / this._ratings.length;
+    
+    return avrRating;
```
Comment (bahriddin) [inline resources.js]: Why you are declaring new variable when you can return the value?

**Review by bahriddin — CHANGES_REQUESTED** [review]

**Review by bahriddin — APPROVED** [review]

**Pull request closed without merging**